SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -299,114 +159,118 @@ void USpatialWorkerConnection::ConnectToLocator()
 
 	Worker_ConnectionFuture* ConnectionFuture = Worker_Alpha_Locator_ConnectAsync(WorkerLocator, &ConnectionParams);
 
+	FinishConnecting(ConnectionFuture);
+}
+
+void USpatialWorkerConnection::FinishConnecting(Worker_ConnectionFuture* ConnectionFuture)
+{
 	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [ConnectionFuture, this]
 	{
-		WorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
-
+		Worker_Connection* NewWorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
 		Worker_ConnectionFuture_Destroy(ConnectionFuture);
-		if (Worker_Connection_IsConnected(WorkerConnection))
+
+		AsyncTask(ENamedThreads::GameThread, [this, NewWorkerConnection]
 		{
+			this->WorkerConnection = NewWorkerConnection;
```

Comment:
```suggestion
			WorkerConnection = NewWorkerConnection;
```

Reply:
I put `this` here to be explicit that this is a member variable and it's being set in the GameThread.